src/flint/types/fmpz_mpoly.pyx

```diff
@@ -977,6 +1066,27 @@ cdef class fmpz_mpoly(flint_mpoly):
         fmpz_mpoly_reduction_primitive_part(res.val, self.val, (<fmpz_mpoly_vec>vec).val, self.ctx.val)
         return res
 
+    def deflation(self):
+        """
+        Compute the deflation of ``self``. See Flint documentation for
+        details. Returns deflated polynomial and the stride vector.
```

Comment:
I went and looked at the Flint documentation which says

>    For each variable `v` let `S_v` be the set of exponents appearing on `v`.
>    Set ``shift[v]`` to `\operatorname{min}(S_v)` and set ``stride[v]`` to `\operatorname{gcd}(S-\operatorname{min}(S_v))`.
>
>    If *A* is zero, all shifts and strides are set to zero.

It is probably best not to refer to the very terse Flint documentation especially when the description there doesn't fully match what this function does. Although we should probably set up intersphinx and see if we can link to the flint docs when we want to refer to them...

Also though I am wondering if handling both shifts and strides is a better thing to do in the python-flint interface. What you have here matches with what existing `deflation` functions in python-flint do I suppose.

Reply:
Actually the shifts thing seems a little odd. I don't think it's needed.

Reply:
> What you have here matches with what existing deflation functions in python-flint do I suppose.

Yep that was my reasoning. Perfectly happy to expose all three methods as well as a sort of "just do whole the deflation" method. Will do so tonight

> It is probably best not to refer to the very terse Flint documentation especially when the description there doesn't fully match what this function does.

That's reasonable. I'll make that change